trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/signing/Signer.java

```diff
@@ -46,9 +50,8 @@ final class Signer
 {
     private static final Logger log = Logger.get(Signer.class);
 
-    static final ZoneId ZONE = ZoneId.of("Z");
-    static final DateTimeFormatter AMZ_DATE_FORMAT = DateTimeFormatter.ofPattern("yyyyMMdd'T'HHmmss'Z'", Locale.US).withZone(ZONE);
-    static final DateTimeFormatter RESPONSE_DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH':'mm':'ss'.'SSS'Z'", Locale.US).withZone(ZONE);
+    @VisibleForTesting
+    static final Duration MAX_PRESIGNED_REQUEST_AGE = Duration.ofDays(7);
```

Comment:
Where does 7 days come from? Can we make this configurable?

Reply:
This is part of the spec: https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html#auth-methods-intro

and https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html

> A presigned URL can be valid for a maximum of seven days because the signing key you use in signature calculation is valid for up to seven days. 

The AWS SDK doesn't even allow you to presign a request with a validity period larger than this: https://github.com/aws/aws-sdk-java/blob/758605bf89ebce8a7d08c4f0829013d88df0513d/aws-java-sdk-core/src/main/java/com/amazonaws/auth/AWS4Signer.java#L622-L629

Reply:
OK - I suggest adding a comment with that link in the code